front/src/applications/operationalStudies/helpers/formatTimetableItemSummaries.ts

```diff
@@ -0,0 +1,104 @@
+import { compact } from 'lodash';
+
+import type {
+  LightRollingStockWithLiveries,
+  SimulationSummaryResult,
+} from 'common/api/osrdEditoastApi';
+import type { TimetableItemWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import type { TimetableItemId, TimetableItemWithTimetableId } from 'reducers/osrdconf/types';
+import { Duration } from 'utils/duration';
+import { jouleToKwh } from 'utils/physics';
+import { formatKmValue } from 'utils/strings';
+import { isPacedTrainResultWithPacedTrainId } from 'utils/trainId';
+import { mapBy } from 'utils/types';
+
+import { isScheduledPointsNotHonored, isTooFast } from '../utils';
+
+/**
+ * Format the timetable items with their simulation summaries
+ */
+const formatTimetableItemSummaries = (
+  timetableItemIds: TimetableItemId[],
+  rawSummaries: Map<TimetableItemId, SimulationSummaryResult>,
+  rawTimetableItems: Map<TimetableItemId, TimetableItemWithTimetableId>,
+  rollingStocks: LightRollingStockWithLiveries[]
+): Map<TimetableItemId, TimetableItemWithDetails> => {
+  const relevantTimetableItems = compact(
+    timetableItemIds.map((timetableItemId) => rawTimetableItems.get(timetableItemId))
+  );
+
+  const items = relevantTimetableItems.map((timetableItem): TimetableItemWithDetails | null => {
+    const rollingStock = rollingStocks.find((rs) => rs.name === timetableItem.rolling_stock_name);
+
+    const timetableItemSummary = rawSummaries.get(timetableItem.id);
+
+    if (!timetableItemSummary) return null;
```

Comment:
This should never happen since you create relevantTimetableItems from rawTimetableItems. You can throw an Exception here
```suggestion
    if (!timetableItemSummary) {
      throw new Error(...)
    }
```

Reply:
Done